Lib/test/test_tools/test_pathfix.py

```diff
@@ -0,0 +1,156 @@
+import unittest, os, shutil, subprocess
+from test.test_tools import import_tool, scriptsdir
+from unittest.mock import patch
+from tempfile import mkdtemp
+import sys
+
+
+class TestPathFixUnit(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.pathfix = import_tool('pathfix')
+
+    test_cases = [
+        (b'#!/usr/bin/python', b'#! /usr/bin/python -f\n'),
+        (b'#!python -f', b'#! /usr/bin/python -f\n'),
+        (b'#! /usr/bin/python -s', b'#! /usr/bin/python -fs\n'),
+        (b'#!/usr/bin/python -f sfj', b'#! /usr/bin/python -f sfj\n'),
+        (b'#!/usr/python -s sfj', b'#! /usr/bin/python -fs sfj\n'),
+    ]
+
+    def test_fixline(self):
+        for line, output in self.test_cases:
+            with self.subTest(line=line, output=output):
+                self.pathfix.add_flag = b'f'
+                self.pathfix.new_interpreter = b'/usr/bin/python'
+                testing_output = self.pathfix.fixline(line)
+                self.assertEqual(testing_output, output)
+
+    test_cases_for_parsing = [
+        (b'#!/usr/bin/python', (b'', b'')),
+        (b'#! /usr/bin/python -f', (b'f', b'')),
+        (b'#!/usr/bin/python -f sfj', (b'f', b'sfj')),
+        (b'#!/usr/bin/python -f sfj af bg', (b'f', b'sfj af bg')),
+    ]
+
+    def test_parse_shebangs(self):
+        for line, output in self.test_cases_for_parsing:
+            with self.subTest(line=line, output=output):
+                self.assertEqual(self.pathfix.parse_shebang(line), output)
+
+    test_cases_without_f_option = [
+        b'#!/usr/bin/python',
+        b'#!python -f',
+        b'#! /usr/bin/python -s',
+        b'#!/usr/bin/python -f sfj',
+        b'#!/usr/python -s sfj',
+    ]
+
+    def test_fixline_without_f_option(self):
+        for line in self.test_cases_without_f_option:
+            with self.subTest(line=line):
+                self.pathfix.new_interpreter = b'/usr/bin/python'
+                self.pathfix.add_flag = None
+                testing_output = self.pathfix.fixline(line)
+                self.assertEqual(testing_output, b'#! ' + self.pathfix.new_interpreter + b'\n')
+
+    input_file = \
```

Comment:
Please keep to [PEP 8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length):

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.